School Library: Basic App UI

IndieCoderMM opened the pull request:
# Creating console UI for the app
The following changes are made in this branch.
- Created `App` class to package all features
- Added required functionalities in `App` class
- Created entry point in `main.rb`
- Refactored code to reduce complexity

IndieCoderMM committed:
Add ui for creating person

IndieCoderMM committed:
Add book creating and listing

IndieCoderMM committed:
Optimize code

franclobo commented on app.rb:
```diff
@@ -0,0 +1,166 @@
+  private
+
+  def create_student(name, age)
+    choice = get_input('Choose classroom ::[M]ath; [C]omputer; [H]istory::', options: %w[m c h])
+    classroom = case choice
+                when 'm'
+                  @classrooms[:math]
+                when 'c'
+                  @classrooms[:computer]
+                when 'h'
+                  @classrooms[:history]
+                end
```
- **[OPTIONAL]** When I see the options to select a classroom I selected the capital letter but it returned an error until I selected the lowercase. Please 🙏🏽 make available to choice "M" or "m" for a better UX.

![image](https://user-images.githubusercontent.com/58642949/225615865-f8c7c240-aec6-4077-b63d-718f13ebdc31.png)

IndieCoderMM replied:
Thanks for your suggestion. 😃 This will definitely improve the overall UX of the app. I will implement it right away.

franclobo reviewed (approved):
Hi @IndieCoderMM, 

# STATUS: APPROVED 🤩 

![git-merge](https://user-images.githubusercontent.com/58642949/225616628-6a2e1be7-0c0c-4de4-9b9b-2b83a8035804.gif)

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter error ✔️ 
- Excelent UI ⭐ ⭐ ⭐ ⭐ ⭐ 
- Excelent UX ⭐ ⭐ ⭐ ⭐ ⭐ 
- Went above and beyond requirements 👏🏽 👏🏽 👏🏽 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

IndieCoderMM committed:
Make UX improvements

- Accept options in both cases
- Add date validation
- Generate classrooms dynamically
